server/filestore.go

```diff
@@ -2131,7 +2132,11 @@ func (fs *fileStore) expireMsgsOnRecover() {
 			break
 		}
 		// Can we remove whole block here?
-		if mb.last.ts <= minAge {
+		// TODO(nat): We can't do this with LimitsTTL as we have no way to know
+		// if we're throwing away real messages or other tombstones without
+		// loading them, so in this case we'll fall through to the "slow path".
+		// There might be a better way of handling this though.
+		if mb.fs.cfg.LimitsTTL <= 0 && mb.last.ts <= minAge {
```

Comment:
LimitsTTL too limiting IMO. Especially if we expand to purge and purgeEx and maybe normal user msg deletes.